cmd/cloudFoundryDeploy.go

```diff
@@ -668,21 +672,101 @@ func deployMta(config *cloudFoundryDeployOptions, mtarFilePath string, command c
 		cfDeployParams = append(cfDeployParams, deployParams...)
 	}
 
-	cfDeployParams = append(cfDeployParams, handleMtaExtensionDescriptors(config.MtaExtensionDescriptor)...)
+	extFileParams, extFiles := handleMtaExtensionDescriptors(config.MtaExtensionDescriptor)
 
-	return cfDeploy(config, cfDeployParams, nil, nil, command)
+	for _, extFile := range extFiles {
+		_, err := fileUtils.Copy(extFile, extFile+".original")
+		if err != nil {
+			return err
+		}
+		err = handleMtaExtensionCredentials(extFile, config.MtaExtensionCredentials)
+		if err != nil {
+			return err
+		}
+	}
+
+	cfDeployParams = append(cfDeployParams, extFileParams...)
+
+	err := cfDeploy(config, cfDeployParams, nil, nil, command)
+
+	for _, extFile := range extFiles {
+		renameError := fileUtils.FileRename(extFile+".original", extFile)
+		if err == nil && renameError != nil {
+			return renameError
```

Comment:
I think it should just continue the loop and maybe log the error. The intention here is to remove the files which contain credentials, no? I guess you could say, if this fails for one file, why should it succeed for the next? But then again, it doesn't hurt to try either.

Reply:
The modified files contains credentials. When restoring the original files without credentials fails this is worth a failure. We don't know which operations are applied to the workspace later on. E.g. the mta extension files might be - by accident - included in some artifacts. I know deployment operations happens normally after creating all the artifacts, but with failing here we are on the safe side. So I would like to recommend to keep it as it is.

Reply:
Ok.